cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -0,0 +1,532 @@
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("container/somedir/subdir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("container/non-existentdir/subdir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("nonexistent-container/somedir"));
+
+  // "Files"
+  ASSERT_OK_AND_ASSIGN(file, fs_->OpenInputFile("container/somefile"));
+  ASSERT_OK_AND_EQ(9, file->GetSize());
+  ASSERT_OK_AND_ASSIGN(buf, file->Read(4));
+  AssertBufferEqual(*buf, "some");
+  ASSERT_OK_AND_EQ(9, file->GetSize());
+  ASSERT_OK_AND_EQ(4, file->Tell());
+
+  ASSERT_OK_AND_ASSIGN(buf, file->ReadAt(2, 5));
+  AssertBufferEqual(*buf, "me da");
+  ASSERT_OK_AND_EQ(4, file->Tell());
+  ASSERT_OK_AND_ASSIGN(buf, file->ReadAt(5, 20));
+  AssertBufferEqual(*buf, "data");
+  ASSERT_OK_AND_ASSIGN(buf, file->ReadAt(9, 20));
+  AssertBufferEqual(*buf, "");
+
+  char result[10];
+  ASSERT_OK_AND_EQ(5, file->ReadAt(2, 5, &result));
+  ASSERT_OK_AND_EQ(4, file->ReadAt(5, 20, &result));
+  ASSERT_OK_AND_EQ(0, file->ReadAt(9, 0, &result));
+
+  // Reading past end of file
+  ASSERT_RAISES(IOError, file->ReadAt(10, 20));
+
+  ASSERT_OK(file->Seek(5));
+  ASSERT_OK_AND_ASSIGN(buf, file->Read(2));
+  AssertBufferEqual(*buf, "da");
+  ASSERT_OK(file->Seek(9));
+  ASSERT_OK_AND_ASSIGN(buf, file->Read(2));
+  AssertBufferEqual(*buf, "");
+  // Seeking past end of file
+  ASSERT_RAISES(IOError, file->Seek(10));
+}
+
+TEST_F(TestAzureFileSystem, OpenOutputStream) {
+  std::shared_ptr<io::OutputStream> stream;
+
+  ASSERT_RAISES(IOError, fs_->OpenOutputStream("container"));
+  ASSERT_RAISES(IOError, fs_->OpenOutputStream("nonexistent-container"));
+  ASSERT_RAISES(IOError, fs_->OpenOutputStream(""));
+  ASSERT_RAISES(IOError, fs_->OpenOutputStream("container/somedir/subdir"));
+  ASSERT_RAISES(IOError, fs_->OpenOutputStream("container/non-existentdir/subdir"));
+  ASSERT_RAISES(IOError, fs_->OpenOutputStream("nonexistent-container/somedir"));
+
+  // Create new empty file
+  ASSERT_OK_AND_ASSIGN(stream, fs_->OpenOutputStream("container/newfile1"));
+  ASSERT_OK(stream->Close());
+  AssertObjectContents(gen2Client_.get(), "container", "newfile1", "");
+
+  // Create new file with 1 small write
+  ASSERT_OK_AND_ASSIGN(stream, fs_->OpenOutputStream("container/newfile2"));
+  ASSERT_OK(stream->Write("some data"));
+  ASSERT_OK(stream->Close());
+  AssertObjectContents(gen2Client_.get(), "container", "newfile2", "some data");
+
+  // Create new file with 3 small writes
+  ASSERT_OK_AND_ASSIGN(stream, fs_->OpenOutputStream("container/newfile3"));
+  ASSERT_OK(stream->Write("some "));
+  ASSERT_OK(stream->Write(""));
+  ASSERT_OK(stream->Write("new data"));
+  ASSERT_OK(stream->Close());
+  AssertObjectContents(gen2Client_.get(), "container", "newfile3", "some new data");
+
+  // Create new file with some large writes
+  std::string s1, s2, s3, s4, s5, expected;
+  s1 = random_string(6000000, /*seed =*/42);  // More than the 5 MB minimum part upload
```

Comment:
```suggestion
  auto s1 = random_string(6000000, /*seed =*/42);  // More than the 5 MB minimum part upload
```

(This is for other variables too.)